data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsGrpcService.java

```diff
@@ -81,7 +78,12 @@ private void processRequest(final ExportMetricsServiceRequest request, final Str
             if (buffer.isByteBuffer()) {
                 buffer.writeBytes(request.toByteArray(), null, bufferWriteTimeoutInMillis);
             } else {
-                buffer.write(new Record<>(request), bufferWriteTimeoutInMillis);
+                Collection<Record<? extends Metric>> metrics;
+                AtomicInteger droppedCounter = new AtomicInteger(0);
```

Comment:
Should we be emitting a metric for this or does the decoder already handle that?

Reply:
Added metrics.